api/tests/app/test_camera_metrics.py

```diff
@@ -0,0 +1,903 @@
+        """
+        camera, camera_2, client, config_sample_path = config_rollback_create_cameras
+        camera_id = camera["id"]
+
+        response = client.get(
+            f"/metrics/cameras/{metric}/weekly?cameras={camera_id}&from_date=2020-09-21&to_date=2020-09-27")
+
+        assert response.status_code == 200
+        assert response.json() == expected
+
+    @pytest.mark.parametrize(
+        "metric,expected",
+        [
+            ("social-distancing", {
+                'detected_objects': [535, 754, 714, 714], 'no_infringement': [416, 622, 555, 555],
+                'low_infringement': [54, 59, 73, 73], 'high_infringement': [38, 56, 55, 55],
+                'critical_infringement': [26, 19, 30, 30],
+                'weeks': ['2020-09-02 2020-09-08', '2020-09-09 2020-09-15', '2020-09-16 2020-09-22',
+                          '2020-09-23 2020-09-29']
+            }),
+            ("face-mask-detections", {
+                'no_face': [88, 85, 106, 85], 'face_with_mask': [310, 519, 445, 519],
+                'face_without_mask': [150, 171, 180, 171],
+                'weeks': ['2020-09-02 2020-09-08', '2020-09-09 2020-09-15', '2020-09-16 2020-09-22',
+                          '2020-09-23 2020-09-29']
+            })
+        ]
+    )
+    @freeze_time("2020-09-30")
+    def test_get_a_weekly_report_properly_weeks_value(self, config_rollback_create_cameras, reports_simulation,
+                                                      metric, expected):
+        """
+        Here we mock datetime.date.today() to a more convenient date set in @freeze_time("2020-09-30")
+        """
+        camera, camera_2, client, config_sample_path = config_rollback_create_cameras
+        camera_id = camera["id"]
+
+        response = client.get(
+            f"/metrics/cameras/{metric}/weekly?cameras={camera_id}&weeks=4")
+
+        assert response.status_code == 200
+        assert response.json() == expected
+
+    @pytest.mark.parametrize(
+        "metric,expected",
+        [
+            ("social-distancing", {
+                'detected_objects': [], 'no_infringement': [], 'low_infringement': [], 'high_infringement': [],
+                'critical_infringement': [], 'weeks': []
+            }),
+            ("face-mask-detections", {'no_face': [], 'face_with_mask': [], 'face_without_mask': [], 'weeks': []})
+        ]
+    )
+    def test_get_a_weekly_report_no_dates_or_week_values(self, config_rollback_create_cameras, reports_simulation,
+                                                         metric, expected):
+        """
+        Here we mock datetime.date.today() to a more convenient date set in @freeze_time("2020-09-30")
+        """
+        camera, camera_2, client, config_sample_path = config_rollback_create_cameras
+        camera_id = camera["id"]
+
+        response = client.get(
+            f"/metrics/cameras/{metric}/weekly?cameras={camera_id}")
+
+        # TODO: Ask if this behaviour is right.
```

Comment:
Hmm I think it should return the number of weeks and fill with zeroes the detections